backends-common/cassandra/src/main/java/org/apache/james/backends/cassandra/utils/ZonedDateTimeRepresentation.java

```diff
@@ -0,0 +1,34 @@
+
+
+package org.apache.james.backends.cassandra.utils;
+
+import java.time.Instant;
+import java.time.ZoneId;
+import java.time.ZonedDateTime;
+
+public class ZonedDateTimeRepresentation {
+
+    private final long msSinceEpoch;
+    private final String serializedZoneId;
```

Comment:
I don't see why you have to store weak type when you can just keep the ZonedDateTime and provide the exact same API.